src/model/EnergyManagementSystemActuator.cpp

```diff
@@ -166,17 +186,25 @@ namespace model {
     }
   }
 
-  EnergyManagementSystemActuator::EnergyManagementSystemActuator(const ModelObject& modelObject, const std::string& actuatedComponentType,
+  EnergyManagementSystemActuator::EnergyManagementSystemActuator(const ModelObject& actuatedComponent, const std::string& actuatedComponentType,
                                                                  const std::string& actuatedComponentControlType, const ThermalZone& thermalZone)
-    : ModelObject(EnergyManagementSystemActuator::iddObjectType(), modelObject.model()) {
+    : ModelObject(EnergyManagementSystemActuator::iddObjectType(), actuatedComponent.model()) {
     OS_ASSERT(getImpl<detail::EnergyManagementSystemActuator_Impl>());
     /** This constructor uses the ThermalZone as the ZoneName field
-      * This is what is expected in E+ for spaceloads in OS that are defined in SpaceTypes and used in Spaces
-  **/
-    bool ok = setActuatedComponent(modelObject);
+      * This is what is expected in E+ for spaceloads in OS that are defined in SpaceTypes and used in Spaces **/
+
+    if (!actuatedComponent.optionalCast<SpaceLoadInstance>()) {
+      remove();
+      LOG_AND_THROW("When using this overloaded EnergyManagementSystemActuator constructor that takes a ThermalZone as the last parameter, the "
+                    "actuated component must be "
+                    "a SpaceLoadInstance, but you supplied "
+                    << actuatedComponent.briefDescription());
+    }
```

Comment:
If you call the ctor that uses the Space / Zone with something else than a SpaceLoadInstance, throw